File: src/main/java/com/tripzin/techminds/entity/WorkOrder.java
package com.tripzin.techminds.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Objects;

@Entity
@Table(name = "work_orders")
@Getter
@Setter
@ToString(exclude = "user")
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WorkOrder {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "wo_number")
    private String woNumber;
    
    @Column(name = "work_type")
    private String workType;
    
    private String client;
    
    @Column(name = "photo_count")
    private Integer photoCount;
    
    private String state;
    
    @Column(name = "client_due_date")
    private LocalDate clientDueDate;
    
    private String updater;
    
    @Column(name = "order_status")
    private String orderStatus;
    
    @Column(name = "remark_category")
    private String remarkCategory;
    
    @Column(name = "start_time")
    private LocalDateTime startTime;
    
    @Column(name = "end_time")
    private LocalDateTime endTime;
    
    @Column(name = "number_of_bids")
    private Integer numberOfBids;
    
    @Column(name = "bid_amount")
    private BigDecimal bidAmount;
    
    @Column(name = "is_rush")
    private boolean isRush;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
    
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        WorkOrder workOrder = (WorkOrder) o;
        return Objects.equals(id, workOrder.id) && 
               Objects.equals(woNumber, workOrder.woNumber);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, woNumber);
    }
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/workOrder.ts
import { User } from '.';

export interface WorkOrder {
  id: number;
  woNumber: string;
  workType: string;
  client: string;
  photoCount?: number;
  state?: string;
  clientDueDate: string;
  updater?: string;
  orderStatus: string;
  remarkCategory?: string;
  startTime?: string;
  endTime?: string;
  numberOfBids?: number;
  bidAmount?: number;
  isRush: boolean;
  user?: User;
  createdAt: string;
  updatedAt: string;
}

export interface WorkOrderStats {
  submitted: number;
  gcSnSubmitted: number;
  rtvFixed: number;
  saved: number;
  totalProcessed: number;
}

export interface WorkOrderFilter {
  woNumber?: string;
  workType?: string;
  client?: string;
  state?: string;
  orderStatus?: string;
  updater?: string;
  isRush?: boolean;
  startDate?: string;
  endDate?: string;
  userId?: number;
}

export interface WorkOrderRequest {
  woNumber: string;
  workType: string;
  client: string;
  photoCount?: number;
  state?: string;
  clientDueDate: string;
  updater?: string;
  orderStatus: string;
  remarkCategory?: string;
  startTime?: string;
  endTime?: string;
  numberOfBids?: number;
  bidAmount?: number;
  isRush: boolean;
  userId?: number;
}

export interface WorkOrdersPageState {
  workOrders: WorkOrder[];
  stats: WorkOrderStats;
  loading: boolean;
  error: string | null;
  page: number;
  totalPages: number;
  rowsPerPage: number;
  sortBy: string;
  sortDir: string;
  filters: WorkOrderFilter;
  selectedWorkOrder: WorkOrder | null;
  openDialog: boolean;
}
